src/particle.android.ts

```diff
@@ -87,4 +113,55 @@ export class Particle implements TNSParticleAPI {
       worker.onmessage = msg => msg.data.success ? resolve(msg.data.result) : reject(msg.data.error);
     });
   }
+
+  private unsubscribe(deviceId: string): void {
+    eventWorker.postMessage({
+      action: "unsubscribe",
+      options: { 
+        deviceId}
+    });
+    cbArr = undefined;
+  }
+
+  private subscribe(deviceId: string, name: string, eventHandler: any): void {
+    if (cbArr == undefined) { 
+      cbArr = {}; 
+    }
+    cbArr[name] = eventHandler;
+    console.dir(cbArr);
+    
+    eventWorker.postMessage({
+      action: "subscribe",
+      options: { 
+        deviceId,
+        name
+      }
+    });
+
+    eventWorker.onmessage = (msg) => {
+      if (msg.data.success) {
+        const d = msg.data.data;
+        // console.log(`${d.name}: ${d.data}`);
+        var cb : (any) => any;
+        cb = cbArr[d.name];
+        if (cb) cb(d.data);
+      }
+    }
+  }
+
+  public isAuthenticated(): Boolean {
+    return ParticleCloudSDK.getCloud().isLoggedIn();
+}
+  
+  public accessToken(): string {
+    return ParticleCloudSDK.getCloud().getAccessToken();
+  }
+
+  public startDeviceSetupWizard(cb:any): void {
+    // stub for startDeviceSetupWizard
```

Comment:
For my info: there's no device setup wizard for Android?

Reply:
There is a device setup wizard available for Android. The device setup library is a separate native dependency - https://github.com/particle-iot/spark-setup-android, more info here https://docs.particle.io/reference/android/#android-device-setup-library. I don't have much experience with android so wasn't comfortable attempting to add that in. I do require it for the project I'm working on though. Is this something that you will be interested in adding?

Reply:
Ah, understood, fair enough! I always strive for x-platform feature parity so I might give it a stab! Not this week though, I'm swamped.

Reply:
.. that being said, that's not blocking for merging this PR of course.

Reply:
Nope, this is perfect, I'll try to merge it asap.